diff --git a/engine/table.py b/engine/table.py
index cda844e..53d81a3 100644
--- a/engine/table.py
+++ b/engine/table.py
@@ -1021,49 +1021,49 @@ class tabengine (ibus.EngineBase):
         # taken and modified from PinYin.py :)
         if self._mode == 1: # refresh mode
             if self._status == u'CN':
-                self._set_property(self._status_property, 'chinese.svg', 'Chinese Mode', 'Switch to English mode')
+                self._set_property(self._status_property, 'chinese.svg', 'Chinese Mode', 'Switch to English mode Shift_R')
             else:
-                self._set_property(self._status_property, 'ibus-table.svg', self._status, 'Switch to English mode')
+                self._set_property(self._status_property, 'ibus-table.svg', self._status, 'Switch to English mode Shift_R')
         else:
-            self._set_property(self._status_property, 'english.svg', 'English Mode', 'Switch to Table mode')
+            self._set_property(self._status_property, 'english.svg', 'English Mode', 'Switch to Table mode Shift_R')
 
         if self._full_width_letter[self._mode]:
-            self._set_property(self._letter_property, 'full-letter.svg', 'Full Letter', 'Switch to half-width letter')
+            self._set_property(self._letter_property, 'full-letter.svg', 'Full Letter', 'Switch to half-width letter Ctrl-Space')
         else:
-            self._set_property(self._letter_property, 'half-letter.svg', 'Half Letter', 'Switch to full-width letter')
+            self._set_property(self._letter_property, 'half-letter.svg', 'Half Letter', 'Switch to full-width letter Ctrl-Space')
 
         if self._full_width_punct[self._mode]:
-            self._set_property(self._punct_property, 'full-punct.svg', 'Full-width Punctuation', 'Switch to half-width punctuation')
+            self._set_property(self._punct_property, 'full-punct.svg', 'Full-width Punctuation', 'Switch to half-width punctuation Ctrl-.')
         else:
-            self._set_property(self._punct_property, 'half-punct.svg', 'Half-width Punctuation', 'Switch to full-width punctuation')
+            self._set_property(self._punct_property, 'half-punct.svg', 'Half-width Punctuation', 'Switch to full-width punctuation Ctrl-.')
 
         if self._editor._py_mode:
-            self._set_property(self._py_property, 'py-mode.svg', 'PinYin Mode', 'Switch to Table mode')
+            self._set_property(self._py_property, 'py-mode.svg', 'PinYin Mode', 'Switch to Table mode Shift_L')
         else:
-            self._set_property(self._py_property, 'tab-mode.svg', 'Table Mode', 'Switch to PinYin mode')
+            self._set_property(self._py_property, 'tab-mode.svg', 'Table Mode', 'Switch to PinYin mode Shift_L')
 
         if self._editor._onechar:
-            self._set_property(self._onechar_property, 'onechar.svg', 'Single Char Mode', 'Switch to phrase mode')
+            self._set_property(self._onechar_property, 'onechar.svg', 'Single Char Mode', 'Switch to phrase mode Ctrl-,')
         else:
-            self._set_property(self._onechar_property, 'phrase.svg', 'Phrase Mode', 'Switch to single char mode')
+            self._set_property(self._onechar_property, 'phrase.svg', 'Phrase Mode', 'Switch to single char mode Ctrl-,')
 
         if self._auto_commit:
-            self._set_property(self._auto_commit_property, 'acommit.svg', 'Direct Commit Mode', 'Switch to normal commit mode, which use space to commit') 
+            self._set_property(self._auto_commit_property, 'acommit.svg', 'Direct Commit Mode', 'Switch to normal commit mode, which use space to commit Ctrl-/')
         else:
-            self._set_property(self._auto_commit_property, 'ncommit.svg', 'Normal Commit Mode', 'Switch to direct commit mode')
+            self._set_property(self._auto_commit_property, 'ncommit.svg', 'Normal Commit Mode', 'Switch to direct commit mode Ctrl-/')
 
         # the chinese_mode:
         if self.db._is_chinese:
             if self._editor._chinese_mode == 0:
-                self._set_property(self._cmode_property, 'sc-mode.svg', 'Simplified Chinese Mode', 'Switch to Traditional Chinese mode')
+                self._set_property(self._cmode_property, 'sc-mode.svg', 'Simplified Chinese Mode', 'Switch to Traditional Chinese mode Ctrl-;')
             elif self._editor._chinese_mode == 1:
-                self._set_property(self._cmode_property, 'tc-mode.svg', 'Traditional Chinese Mode', 'Switch to Simplify Chinese first Big Charset Mode')
+                self._set_property(self._cmode_property, 'tc-mode.svg', 'Traditional Chinese Mode', 'Switch to Simplify Chinese first Big Charset Mode Ctrl-;')
             elif self._editor._chinese_mode == 2:
-                self._set_property(self._cmode_property, 'scb-mode.svg', 'Simplified Chinese First Big Charset Mode', 'Switch to Traditional Chinese first Big Charset Mode')
+                self._set_property(self._cmode_property, 'scb-mode.svg', 'Simplified Chinese First Big Charset Mode', 'Switch to Traditional Chinese first Big Charset Mode Ctrl-;')
             elif self._editor._chinese_mode == 3:
-                self._set_property(self._cmode_property, 'tcb-mode.svg', 'Traditional Chinese First Big Charset Mode', 'Switch to Big Charset Mode')
+                self._set_property(self._cmode_property, 'tcb-mode.svg', 'Traditional Chinese First Big Charset Mode', 'Switch to Big Charset Mode Ctrl-;')
             elif self._editor._chinese_mode == 4:
-                self._set_property(self._cmode_property, 'cb-mode.svg', 'Big Chinese Mode', 'Switch to Simplified Chinese Mode')
+                self._set_property(self._cmode_property, 'cb-mode.svg', 'Big Chinese Mode', 'Switch to Simplified Chinese Mode Ctrl-;')
 
         # use buildin method to update properties :)
         map (self.update_property, self.properties)
